rzup/install

```diff
@@ -1,54 +1,214 @@
 #!/usr/bin/env bash
+# Copyright 2024 RISC Zero, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
 
-# Adapted from https://github.com/foundry-rs/foundry/blob/master/foundryup/install
+set -euo pipefail
```

Comment:
Currently getting a couple of `unbound variable` errors due to the `-u` flag. 
The changes @flaub suggested should fix this.